### PR TITLE
Improve text input a11y for screen readers

### PR DESCRIPTION
**Pull request by flozia**
## Description

Improve the accessibility of text inputs for screen readers.

## Reference

- #4555
- #4556 

## Checklist
    
- [x] My code follows the style guidelines for this project
- [x] I have not added any packages that contain high risk or unknown licenses (GPL,  LGPL, MPL, etc. consult with DevOps if in question)
- [x] I have performed a self review of my own code
- [x] I have commented my code PARTICULARLY in hard to understand areas
- [x] I have added thorough tests where needed

### CODE REVIEW

File: nebula/ui/components/forms/VPNTextField.qml [hunk 1]
```diff
@@ -10,50 +10,69 @@ import Mozilla.VPN 1.0
...
+        vpnFlickable.ensureVisible(textField);
+    }
+    // This is a workaround to make VoiceOver on macOS work.
+    // After gaining initial focus or typing in TextField the screen reader
+    // fails to narrae any accessible content and action. After regaining active
+    // focus the screen reader keeps working as expected.
+    onTextChanged: {
+        textField.focus = false;
+        textField.forceActiveFocus();
+    }
```
Comment (flozia) [inline nebula/ui/components/forms/VPNTextField.qml:56]: That’s definitely not optimal, but …

File: nebula/ui/components/forms/VPNTextField.qml [hunk 2]
```diff
@@ -10,50 +10,69 @@ import Mozilla.VPN 1.0
...
     property bool forceBlurOnOutsidePress: true
     property alias _placeholderText: centeredPlaceholderText.text
+    property string _accessibleName: _placeholderText
+    property string _accessibleDescription: ""
 
     id: textField
 
+    Accessible.name: _accessibleName
+    Accessible.description: _accessibleDescription
+    Accessible.focused: textField.focus
```
Comment (flozia) [inline nebula/ui/components/forms/VPNTextField.qml:24]: … does fix issue #4556 at least for VoiceOver on macOS in combination with setting `Accessible.focused`.

File: nebula/ui/components/forms/VPNSearchBar.qml [hunk 3]
```diff
@@ -16,34 +16,41 @@ ColumnLayout {
...
     VPNTextField {
-        // TODO Add strings for Accessible.description, Accessible.name
-
         id: searchBar
 
-        background: VPNInputBackground {}
-        leftInset: 48
-        leftPadding: 48
-        onActiveFocusChanged: if (focus && vpnFlickable.ensureVisible) vpnFlickable.ensureVisible(searchBar)
+        Accessible.editable: false
```
Comment (flozia) [inline nebula/ui/components/forms/VPNSearchBar.qml:27]: Fixes issue #4555.

**Review by flozia — COMMENTED** [review]
Left comments for the relevant changes. The rest of the changes are mostly formatting changes for readability reasons.

File: nebula/ui/components/forms/VPNTextField.qml (hunk 1 above)
Comment (MattLichtenstein) [inline nebula/ui/components/forms/VPNTextField.qml:56]: Should we apply this to VPNTextArea as well? And maybe prevent it from being used by VPNPasswordInput?

**Review by MattLichtenstein — APPROVED** [review]
Clever solution, and I think it does more good than harm so I say ship it.

Only problem I could see arising in the future is if there is some circumstance where we are changing text in a textfield through some other means than typing, but we can cross that bridge if its ever even built!

File: nebula/ui/components/forms/VPNTextArea.qml [hunk 4]
```diff
@@ -42,36 +44,52 @@ Item {
...
+            Keys.onTabPressed: nextItemInFocusChain().forceActiveFocus(Qt.TabFocusReason)
+            onTextChanged: {
+                handleOnTextChanged(textArea.text);
+
+                // This is a workaround to make VoiceOver on macOS work.
+                // After gaining initial focus or typing in TextArea the screen reader
+                // fails to narrate any accessible content and action. After regaining
+                // active focus the screen reader keeps working as expected.
+                textArea.focus = false;
+                textArea.forceActiveFocus();
```
Comment (flozia) [inline nebula/ui/components/forms/VPNTextArea.qml:78]: Applied the fix for `VPNTextArea` as suggested by @MattLichtenstein as well.

File: nebula/ui/components/forms/VPNPasswordInput.qml [hunk 5]
```diff
@@ -15,6 +15,14 @@ VPNTextField {
     property alias placeholder: passwordInput._placeholderText
 
     id: passwordInput
+
+    Accessible.ignored: charactersMasked
+    Accessible.passwordEdit: charactersMasked
+    // In order for screen readers to respect the `Accessible.ignored` property
+    // `Accessible.role` has to be set to `EditableText`. For more infos see
+    // QTBUG-82433: https://bugreports.qt.io/browse/QTBUG-82433.
+    Accessible.role: Accessible.EditableText
```
Comment (flozia) [inline nebula/ui/components/forms/VPNPasswordInput.qml:24]: Do not narrate content when the password input is masked.

**Review by flozia — COMMENTED** [review]
Added a few more changes:
- Applied the fix for #4556 also to `VPNTextArea`
- Prevent screen readers from narrating password inputs if masked
- Used the opportunity to do a bit more formatting in `VPNTextArea` and `VPNPasswordInput`

**Review by MattLichtenstein — APPROVED** [review]
👌

**Pull request merged**